src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -1612,6 +1612,7 @@ private void DoCreateRunspace(string initialCommand, bool skipProfiles, bool sta
                     OpenConsoleRunspace(consoleRunspace, staMode);
                 }
 
+                Runspace.PrimaryRunspace = consoleRunspace;
```

Comment:
It seems like PrimaryRunspace gets set again within OpenConsoleRunspace if it fails to open the first time.  But I agree it makes more sense to set it here.